mycroft/stt/deepspeech_stt.py

```diff
@@ -0,0 +1,131 @@
+        self.lm = self.config.get("lm", join(dirname(__file__),
+                                             "deepspeech",
+                                             "lm.binary"))
+        self.alphabet = self.config.get("alphabet", join(dirname(__file__),
+                                                         "deepspeech",
+                                                         "alphabet.txt"))
+        self.trie = self.config.get("trie", join(dirname(__file__),
+                                                 "deepspeech",
+                                                 "trie"))
+        self.downloaded = False
+        self.dl = None
+        auto_dl = self.config.get("auto_download", False)
+        if self.is_ready(not auto_dl):
+            self.load_model()
+        elif auto_dl:
+            LOG.info("Downloading model")
+            self.download()
+            while not self.downloaded:
+                sleep(1)
+
+            if self.is_ready(True):
+                self.load_model()
+
+    def load_model(self):
+        LOG.info('Loading model from file %s' % (self.model))
+        model_load_start = timer()
+        self.ds = self.Model(self.model, self.N_FEATURES, self.N_CONTEXT,
+                             self.alphabet, self.BEAM_WIDTH)
+
+        model_load_end = timer() - model_load_start
+        LOG.debug('Loaded model in %0.3fs.' % (model_load_end))
+
+        LOG.info('Loading language model from files %s %s' % (
+            self.lm, self.trie))
+        lm_load_start = timer()
+        self.ds.enableDecoderWithLM(self.alphabet, self.lm, self.trie,
+                                    self.LM_WEIGHT,
+                                    self.WORD_COUNT_WEIGHT,
+                                    self.VALID_WORD_COUNT_WEIGHT)
+        lm_load_end = timer() - lm_load_start
+        LOG.debug('Loaded language model in %0.3fs.' % (lm_load_end))
+
+    def is_ready(self, is_critical=False):
+        try:
+            if not exists(self.model):
+                raise AssertionError(
+                    self.model + " does not exist, download a "
+                                 "pre-trained model with \n wget -O -" +
+                    DeepSpeechSTT.URL + "| tar xvfz -")
+            if not exists(self.lm):
+                raise AssertionError("language model does not exist")
+
+            if not exists(self.alphabet):
+                raise AssertionError(
+                    "alphabet configuration file does not exist")
+
+            if not exists(self.trie):
+                raise AssertionError("language model trie does not exist")
+        except Exception as e:
+            LOG.error(e)
+            if is_critical:
+                raise
+            return False
+        return True
+
+    def download(self):
+        LOG.info("starting model download")
+        target_folder = join(dirname(__file__), "deepspeech")
+        self.dl = download(DeepSpeechSTT.URL, target_folder, self._extract)
```

Comment:
why is this async? Can we just use a sync version instead?

Reply:
i just used the official download util, this way we would get any updates to the download code instead of me re implementing it